### PR TITLE
[Header] Add Id to the container header tag

### PR DESCRIPTION
**Pull request by StevenTCramer**
# Pull Request

## 📖 Description

The Id property was not being set on FluentHeader

## 📑 Test Plan

Added FluentHeaderTests which includes
* FluentHeader_Default
* FluentHeader_AdditionalProperties
* FluentHeader_Id

### General

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fluentui-blazor/blob/master/docs/contributing.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have added [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for my new compontent
- [x] I have modified an existing component
- [x] I have validated the [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for an existing component

### CODE REVIEW

File: tests/Core/Header/FluentHeaderTests.razor [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    public FluentHeaderTests()
+    {
+        JSInterop.Mode = JSRuntimeMode.Loose;
+        Services.AddSingleton(LibraryConfiguration);
+    }
+    [Fact]
+    public void FluentHeader_Default()
+    {
+        //Arrange
+        var childContent = "<b>render me</b>";
```
Comment (dvoituron) [inline tests/Core/Header/FluentHeaderTests.razor:18]: Please, don't use variable in Unit Tests, but set the value directly in ´Add´ and ´AddChildren´ methods.

**Review by dvoituron — CHANGES_REQUESTED** [review]

File: tests/Core/Header/FluentHeaderTests.razor (hunk 1 above)
Comment (StevenTCramer) [reply tests/Core/Header/FluentHeaderTests.razor:18]: Ok so these other 28 all need updated as well.  I might do that this weekend. In another PR.  Can we just approve this one as is and I will fix them all at once?
image

Comment (StevenTCramer) [reply tests/Core/Header/FluentHeaderTests.razor:18]: Searching I could find only one test that sets it directly. If you want to change this, I can fix them all at once. But not until the weekend.

Comment (StevenTCramer) [reply tests/Core/Header/FluentHeaderTests.razor]: But I did update this one.

Comment (vnbaaij) [reply tests/Core/Header/FluentHeaderTests.razor]: I'll approve this one, but please update the others as well

**Review by vnbaaij — APPROVED** [review]

File: tests/Core/Header/FluentHeaderTests.razor (hunk 1 above)
Comment (dvoituron) [reply tests/Core/Header/FluentHeaderTests.razor]: I'm not necessarily asking you to modify our old code, we can do that again during a future refactoring session. But you can take care of it :-)
My comment was more about your code, which could use better practices than what we did at the start :-)

**Review by dvoituron — APPROVED** [review]

**Pull request merged**